sklearn/metrics/_dist_metrics.pxd

```diff
@@ -60,9 +59,25 @@ cdef class DistanceMetric:
     cdef DTYPE_t dist(self, const DTYPE_t* x1, const DTYPE_t* x2,
                       ITYPE_t size) nogil except -1
 
-    cdef DTYPE_t rdist(self, DTYPE_t* x1, DTYPE_t* x2,
+    cdef DTYPE_t rdist(self, const DTYPE_t* x1, const DTYPE_t* x2,
                        ITYPE_t size) nogil except -1
 
+    cdef DTYPE_t sparse_dist(
```

Comment:
I think we'll only support the csr format, which is the most appropriate for pairwise distance computation, so i'd name it csr_dist and csr_rdist instead to prevent any confusion

Reply:
https://github.com/jjerphan/scikit-learn/pull/4 by @mbatoul uses a proper naming.

I'd be in favor or reverting the patch for fused {dense, sparse} dataset pairs in this PR and tackle it the support in a subsequent one.